harness/determined/core/_checkpoint.py

```diff
@@ -222,32 +235,27 @@ def _report_checkpoint(
         """
         After having uploaded a checkpoint, report its existence to the master.
         """
-
         resources = resources or {}
         metadata = metadata or {}
-        required = {"latest_batch"}
-        allowed = required.union({"framework", "format", "total_records", "total_epochs"})
-        missing = [k for k in required if k not in metadata]
-        extra = [k for k in metadata.keys() if k not in allowed]
-        if missing:
-            raise ValueError(
-                "metadata for reported checkpoints, in the current implementation, requires all of "
-                f"the following items that have not been provided: {missing}"
-            )
-        if extra:
+
+        if "latest_batch" not in metadata:
             raise ValueError(
-                "metadata for reported checkpoints, in the current implementation, cannot support "
-                f"the following items that were provided: {extra}"
+                "metadata for reported checkpoints, in the current implementation, requires a "
+                "'latest_batch' item, which has not been provided"
             )
 
-        body = {
-            "uuid": storage_id,
-            "resources": resources,
-            **self._static_metadata,
-            **metadata,
-        }
+        ckpt = bindings.v1Checkpoint(
+            allocationId=self._allocation_id,
+            metadata=metadata,
+            resources={k: str(v) for k, v in resources.items()},
+            taskId=self._task_id,
+            training=bindings.v1CheckpointTrainingMetadata(),
+            uuid=storage_id,
+            reportTime=datetime.now(timezone.utc).isoformat(),
+            state=bindings.determinedcheckpointv1State.STATE_COMPLETED,
+        )
+        bindings.post_ReportCheckpoint(self._session, body=ckpt)
         logger.info(f"Reported checkpoint to master {storage_id}")
-        self._session.post(self._api_path, data=det.util.json_encode(body))
```

Comment:
did you figure out a way to use our custom `json_encode()` with our bindings?  Otherwise, this will cause problems when users try to put e.g. metrics into their metadata (which seems pretty reasonable to me).

Reply:
no I am not using it. What did I miss?
```
            checkpoint_metadata = {"latest_batch": batch, "metrics": {"x": x}}
            with core_context.checkpoint.store_path(checkpoint_metadata) as (path, uuid):
                save_state(x, batch, trial_id, path)
```
^ this works.
What wouldn't?

Reply:
There's quite a few corner cases we've had to add to `json_encode()` in the past.  Some of the more important ones are np arrays, nans, and infs.

Reply:
ok, no I did not take care of that. We agreed to carve it out and address it later: https://determinedai.atlassian.net/browse/DET-7166